src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,88 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserAgeException;
+import core.basesyntax.exception.UserLoginExistsException;
+import core.basesyntax.exception.UserLoginLengthException;
+import core.basesyntax.exception.UserPasswordLengthException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int USER_LOGIN_LENGTH_MIN = 6;
+    private static final int USER_PASSWORD_LENGTH_MIN = 6;
+    private static final int USER_AGE_MIN = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        String userLogin = user.getLogin();
```

Comment:
Let's make your implementation more elegant 
```suggestion
        if (user == null) {
            throw new RegistrationException("The user doesn't exist");
        }
        validateLogin(user);
        validatePassword(user);
        validateAge(user);
        return storageDao.add(user);
```

Where example of method

```
private void validateLogin(User user) {
        if (user.getLogin() == null) {
            throw new RegistrationException("Login can't be null.");
        }
        if (user.getLogin().equals("")) {
            throw new RegistrationException("Your login can't be an empty line.");
        }
        if (user.getLogin().length() < MIN_LENGTH) {
            throw new RegistrationException("Your login should contain %s or more symbols."
                    .formatted(MIN_LENGTH));

        }
        if (storageDao.get(user.getLogin()) != null) {
            throw new RegistrationException("User with login %s already exists."
                    .formatted(user.getLogin()));
        }
    }

```